### PR TITLE
Adding NullBool type.

### PR DESCRIPTION
**Pull request by umerazad**
Added a NullBool type. Plus a minor UT change to increase coverage for NullInt64 type.

### CODE REVIEW

File: null_bool.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// NullBool is a type that holds an optional boolean value.
+type NullBool struct {
+	sql.NullBool
+}
+
+// NewNullBool returns a new NullBool object that is initialized to the
+// specified value.
+func NewNullBool(value bool) NullBool {
+	return NullBool{
+		NullBool: sql.NullBool{
```
Comment (rkuris) [inline null_bool.go]: nit: `NullBool:` is optional. While it is the name, this is an anonymous member.

File: null_bool.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	}
+
+	switch unknown.(type) {
+	case bool:
+		err = json.Unmarshal(data, &i.Bool)
+	case nil:
+		i.Valid = false
+		return nil
+	default:
+		err = fmt.Errorf("failed to unmarshal %v into NullBool", reflect.TypeOf(unknown).Name())
```
Comment (rkuris) [inline null_bool.go]: nit: use %T which does the reflect for you for free:

`fmt.Errorf("failed to unmarshal type %T into NullBool", unknown)`

File: null_int64_test.go [hunk 3]
```diff
@@ -3,6 +3,8 @@ package dosa
 import (
 	"testing"
 
+	"encoding/json"
```
Comment (rkuris) [inline null_int64_test.go]: nit: was this import required? Looks unneeded?

File: null_bool.go (hunk 1 above)
Comment (umerazad) [inline null_bool.go]: Fixed.

File: null_bool.go (hunk 2 above)
Comment (umerazad) [inline null_bool.go]: Fixed.

File: null_int64_test.go (hunk 3 above)
Comment (umerazad) [inline null_int64_test.go]: Its used on line 85.

**Pull request merged**